Tribler/community/tunnel/hidden_community.py

```diff
@@ -294,17 +374,22 @@ def check_key_request(self, messages):
 
     def on_key_request(self, messages):
         for message in messages:
-            # if we have received this message over a socket, we need to forward it
             if not message.source.startswith(u"circuit_"):
+                # The intropoint receives the message over a socket, and forwards it to the seeder
                 self._logger.debug("On key request: relay key request")
+                cache = self.request_cache.add(KeyRelayCache(self, message.payload.identifier, message.payload.circuit_id))
+                meta = self.get_meta_message(u'key-request')
+                message = meta.impl(distribution=(self.global_time,), payload=(message.payload.circuit_id, cache.number, message.payload.info_hash))
                 relay_circuit = self.intro_point_for[message.payload.info_hash]
-                relay_circuit.tunnel_data(message.candidate.sock_addr, TUNNEL_PREFIX + message.packet)
```

Comment:
I'm not sure if this is a correct change. My understanding was that message.candidate.sock_addr was the ip/port of the exit-socket of the previous tunnel.
Hence, if we backpropagate it to that ip/port, it would automatically be forwarded back to the originator.